static/js/src/PurchaseModal/components/PaymentMethodForm.jsx

```diff
@@ -0,0 +1,318 @@
+          />
+          <Field
+            as={RadioInput}
+            name="buyingFor"
+            value="organisation"
+            label="An organisation"
+            defaultChecked={values.buyingFor === "organisation"}
+          />
+        </div>
+      </FormRow>
+      <Field
+        as={Input}
+        type="text"
+        id="organisationName"
+        name="organisationName"
+        disabled={values.buyingFor === "myself"}
+        label="Organisation:"
+        stacked
+        validate={validateOrganisationName}
+        error={touched?.organisationName && errors?.organisationName}
+      />
+      <Field
+        as={Input}
+        type="text"
+        id="address"
+        name="address"
+        label="Address:"
+        stacked
+        validate={validateRequired}
+        error={touched?.address && errors?.address}
+      />
+      <Field
+        as={Input}
+        type="text"
+        id="postalCode"
+        name="postalCode"
+        label="Postal code:"
+        stacked
+        validate={validateRequired}
+        error={touched?.postalCode && errors?.postalCode}
+      />
+      <Field
+        as={Select}
+        id="country"
+        name="country"
+        options={countries}
+        label="Country/Region:"
+        stacked
+        validate={validateRequired}
+        error={touched?.country && errors?.country}
+      />
+      <Field
+        as={Input}
+        type="text"
+        id="city"
+        name="city"
+        label="City:"
+        stacked
+        validate={validateRequired}
+        error={touched?.city && errors?.city}
+      />
+      {values.country === "US" && (
+        <Field
+          as={Select}
+          id="usStates"
+          name="usState"
+          options={USStates}
+          label="State:"
+          stacked
+          validate={validateUSState}
+          error={touched?.usState && errors?.usState}
+        />
+      )}
+      {values.country === "CA" && (
+        <Field
+          as={Select}
+          id="caProvinces"
+          name="caProvince"
+          options={caProvinces}
+          label="Province:"
+          stacked
+          validate={validatecaProvince}
+          error={touched?.caProvince && errors?.caProvince}
+        />
+      )}
+      {vatCountries.includes(values.country) && (
+        <Field
+          as={Input}
+          type="text"
+          id="VATNumber"
+          name="VATNumber"
+          label="VAT number:"
+          stacked
+          help="e.g. GB 123 1234 12 123 or GB 123 4567 89 1234"
+          error={touched?.VATNumber && errors?.VATNumber}
+        />
+      )}
+
+      <Row className="u-no-padding u-sv1">
+        <Col size="8" emptyLarge="5">
+          <p>You will have a chance to review on the next step...</p>
+        </Col>
+      </Row>
+    </Form>
+  );
+}
+
+PaymentMethodForm.propTypes = {
+  setCardValid: PropTypes.func.isRequired,
+};
+
+export default PaymentMethodForm;
```

Comment:
Have you tried converting this file to TS? And if so, did you have any issues that were blocking you?

Reply:
Done